atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractAtlasDbKeyValueServiceTest.java

```diff
@@ -660,21 +661,104 @@ public void testGetRangeOfTimestampsThrowsOnError() {
 
     @Test
     public void testCannotModifyValuesAfterWrite() {
+        Cell cell = Cell.create(row0, column0);
         byte[] data = new byte[1];
-        byte[] unmodifiedData = Arrays.copyOf(data, data.length);
+        byte[] unmodifiedData = writeToCell(cell, data);
```

Comment:
might want to assert that the `data` and `unmodifiedData` the equal, but not the same reference:

```
        assertThat(unmodifiedData, is(data));
        assertThat(unmodifiedData, not(sameInstance(data)));
```